core/services/directrequest/delegate.go

```diff
@@ -54,13 +70,23 @@ func (d *Delegate) ServicesForSpec(spec job.Job) (services []job.Service, err er
 		return
 	}
 
+	minConfirmations := d.config.MinRequiredOutgoingConfirmations()
+	if concreteSpec.NumConfirmations > minConfirmations {
+		minConfirmations = concreteSpec.NumConfirmations
+	}
+
 	logListener := &listener{
-		logBroadcaster: d.logBroadcaster,
-		oracle:         oracle,
-		pipelineRunner: d.pipelineRunner,
-		db:             d.db,
-		pipelineORM:    d.pipelineORM,
-		spec:           *spec.PipelineSpec,
+		logBroadcaster:   d.logBroadcaster,
+		headBroadcaster:  d.headBroadcaster,
+		oracle:           oracle,
+		pipelineRunner:   d.pipelineRunner,
+		db:               d.db,
+		pipelineORM:      d.pipelineORM,
+		spec:             *spec.PipelineSpec,
+		mbLogs:           utils.NewMailbox(d.config.MailboxCapacity()),
```

Comment:
What failure scenario would cause us to exceed this mailbox limit?

Reply:
in the current implementation here - it would start skipping logs if there were too many relevant logs for the same job (> MailboxCapacity) in each block.
However, that will change in the new LB implementation - LB is able to keep all logs in memory while they wait to be confirmed

Reply:
added a comment in the code